Extra/Settings/SettingGui.js

```diff
@@ -357,14 +362,120 @@ class SettingGui {
      * @param {String} setting internal name of the setting to control
      * @param {String} defau Default value
      */
-    addString(label, setting, defau) {
+     addString(label, setting, defau) {
         let textBox = this.addSidebarElement(new TextBox().setText(this.defaultSettings[setting] ?? defau).addEvent(new SoopyContentChangeEvent().setHandler((val, prev, cancelFun) => {
             this.changed(setting, val)
         })), 0.625, 0.2, 0.05)
 
         return [textBox, this.addSidebarElement(new SoopyTextElement().setText("§0" + label).setMaxTextScale(2), 0.1, 0.35)]
     }
 
+    /**
+     * 
+     * @param {String} label The text to go to the left of the slider
+     * @param {String} setting internal name of the setting to control
+     * @param {Object[]} defau Default value
+     */
+    addColorSelector(label, setting, defau) {
+        // Slider 0
+        let slider0 = new Slider().setValue(this.defaultSettings[setting][0] ?? defau).setMin(0).setMax(255).addEvent(new SoopyContentChangeEvent().setHandler((val, prev, cancelFun) => {
+            this.changedArr(setting, 0, Math.round(val))
+            numberT0.setText(Math.round(val).toString())
+        }))
+
+        let numberT0 = new NumberTextBox().setText((this.defaultSettings[setting][0] ?? defau).toString())
+
+        numberT0.isNumber = (val) => {
+            if (val.includes(".")) return false
+            val = "" + val; //coerce num to be a string
+            return !isNaN(val) && !isNaN(parseInt(val));
+        }
+
+        numberT0.text.addEvent(new SoopyContentChangeEvent().setHandler((val, prev, cancelFun) => {
+            if (!val) return
+
+            this.changedArr(setting, 0, parseInt(val))
+            slider0.setValue(parseInt(val))
+        }))
+        this.addSidebarElement(slider0, 0.26, 0.09, 0.05).setLore(["red"])
+        this.addSidebarElement(numberT0, 0.35, 0.08, 0.05).setLore(["red"])
+        
+        // Slider 1
+        let slider1 = new Slider().setValue(this.defaultSettings[setting][1] ?? defau).setMin(0).setMax(255).addEvent(new SoopyContentChangeEvent().setHandler((val, prev, cancelFun) => {
+            this.changedArr(setting, 1, Math.round(val))
+            numberT1.setText(Math.round(val).toString())
+        }))
+
+        let numberT1 = new NumberTextBox().setText((this.defaultSettings[setting][1] ?? defau).toString())
+
+        numberT1.isNumber = (val) => {
```

Comment:
Also it would be nice if u extracted the isNumber function out to a variable at the bottom of the file so you could replace all these with just numberT1.isNumber = isNumber